### PR TITLE
all: adding codeowners

### PR DESCRIPTION
**Pull request by viktor-f**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [x] personal data beyond what is necessary for interacting with this pull request, nor
> - [x] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [ ] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [x] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### What does this PR do / why do we need this PR?

This is a first draft of adding codeowners to this repo.
There are probably some things that could be changed or something that was missed. Please at least look at your goto areas to see that they seem sensible. We would also appreciate comments on the skipped files to see if they should be given to any area.

- Fixes #

#### Checklist

- [ ] Proper commit message prefix on all commits
  
- Change checks:
  - [ ] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
  - [ ] The change upgrades CRDs
  - [ ] The change updates the config *and* the schema
- Documentation checks:
  - [ ] The [public documentation](https://github.com/elastisys/compliantkubernetes) required no updates
  - [ ] The [public documentation](https://github.com/elastisys/compliantkubernetes) required an update - [link to change](set-me\)
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

File: .github/CODEOWNERS
Comment (Ajarmar) [inline .github/CODEOWNERS:157]: Should be continuous delivery goto?

File: .github/CODEOWNERS
Comment (AlbinB97) [inline .github/CODEOWNERS:3]: Shouldnt this be @elastisys/goto-scripts ?

File: .github/CODEOWNERS
Comment (Xartos) [inline .github/CODEOWNERS:1]: This doesn't seem to be allowed. Or is it just a bad parser?

Also, do we want to have sub teams with this instead of having it as a parent group? Then the goto-pipeline would automatically include all in all the subteams

File: .github/CODEOWNERS
Comment (Xartos) [inline .github/CODEOWNERS:3]:
```suggestion
/bin/ @elastisys/goto-scripts
```

**Review by Ajarmar — COMMENTED** [review]
It seems there are a lot of errors in the CODEOWNERS file, but I imagine it is something sysadmin needs to fix? Specifically, teams not having write access to the repo/not being publicly visible, and team names with `/` in them look like they don't work correctly.

File: .github/CODEOWNERS
Comment (Ajarmar) [inline .github/CODEOWNERS:83]:
```suggestion
/helmfile.d/stacks/thanos.yaml @elastisys/goto-monitoring-stack
```

File: .github/CODEOWNERS
Comment (Ajarmar) [inline .github/CODEOWNERS:68]:
```suggestion
/helmfile.d/stacks/fluentd.yaml @elatisys/goto-logging-stack
```

File: .github/CODEOWNERS
Comment (Xartos) [inline .github/CODEOWNERS:7]: These doesn't seem to be public enough :thinking:

File: .github/CODEOWNERS
Comment (Xartos) [inline .github/CODEOWNERS:207]: Do we need to have this if we have them for `/scripts/*`?

File: .github/CODEOWNERS
Comment (Xartos) [inline .github/CODEOWNERS:209]: Could this be changed to
```suggestion
/scripts/ @elastisys/goto-scripts
```

File: .github/CODEOWNERS
Comment (Xartos) [inline .github/CODEOWNERS:206]: We should probably rename this folder, or create a subfolder

```suggestion
/scripts/restore/harbor/ @elastisys/goto-container-registry
```

**Comment by viktor-f** [comment]
As people have noted there seem to be issues with the goto teams not being publicly visible and not being able to have `/` in the name.

@Xartos or @robinelastisys 
Could you look into making them public and renaming "goto-ingress/auth" to "goto-ingress-auth" and "goto-pipeline/qa/release" to "goto-pipeline-qa-release" (or something similar)?

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:218]: I added that goto for `.github/` and then thought that this was an adjacent file. But the reasoning was not very concrete :sweat_smile: 
If you want to remove or change it then I'm very open for suggestions.

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:207]: Yes, as far as I understand. This is from the codeowner documentation:

> The `docs/*` pattern will match files like
 `docs/getting-started.md` but not further nested files like
 `docs/build-app/troubleshooting.md`.
docs/*  docs@example.com

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:209]: It will give a bit of a different meaning, check my comment to your previous comment.

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:206]: Yeah I was thinking about that as well. I was just not sure if I wanted to do that in this PR and if that had implications on other things as well.

File: .github/CODEOWNERS
Comment (aarnq) [inline .github/CODEOWNERS:20]: Is this chart even used nowadays? :thinking:

File: .github/CODEOWNERS
Comment (aarnq) [inline .github/CODEOWNERS:208]:
```suggestion
/scripts/report/ @elastisys/goto-pipeline-qa-release
```

**Comment by Xartos** [comment]
> Could you look into making them public

As far as I can see they are as public as they can be :thinking: @robinelastisys Do you know some other way of making them even more public

File: .github/CODEOWNERS
Comment (aarnq) [reply .github/CODEOWNERS:213]: You should be able to set us for `/scripts/` before other entries in script, and the more specific ones will override the generic one.

From the docs:
```
# In this example, @octocat owns any file in the `/apps`
# directory in the root of your repository except for the `/apps/github`
# subdirectory, as this subdirectory has its own owner @doctocat
/apps/ @octocat
/apps/github @doctocat
```

**Comment by aarnq** [comment]
> > Could you look into making them public
> 
> As far as I can see they are as public as they can be 🤔 @robinelastisys Do you know some other way of making them even more public

Maybe:
> https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax
> Users and teams must have explicit write access to the repository, even if the team's members already have access.

**Comment by Xartos** [comment]
> > > Could you look into making them public
> > 
> > 
> > As far as I can see they are as public as they can be 🤔 @robinelastisys Do you know some other way of making them even more public
> 
> Maybe:
> 
> > [docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax)
> > Users and teams must have explicit write access to the repository, even if the team's members already have access.

Yea, that's probably correct. But that feels like a bunch of work that I don't want to manage manually. I'll take a look if this can be managed in a better way. We will probably want to manage the teams in a better way anyway so I think that is a good opportunity to fix that

File: .github/CODEOWNERS
Comment (Ajarmar) [reply .github/CODEOWNERS:222]: I think it doesn't really apply to any goto area, probably makes most sense to remove it

File: .github/CODEOWNERS
Comment (simonklb) [inline .github/CODEOWNERS:231]: This should be owned by `goto-scripts` IMO as it declares the dependencies for running the scripts in `/bin`.

File: .github/CODEOWNERS
Comment (simonklb) [inline .github/CODEOWNERS:227]: QA?

File: .github/CODEOWNERS
Comment (simonklb) [inline .github/CODEOWNERS:230]: I wonder if this should be maintained by public docs?

**Review by robinelastisys — APPROVED** [review]

File: .github/CODEOWNERS
Comment (Xartos) [inline .github/CODEOWNERS:1]: Permission wise, all should be fixed. However these needs do be fixed before we can merge

```suggestion
/.github/ @elastisys/goto-pipeline-qa-release
```

**Review by Xartos — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS]: I think that makes some sense, added it.
However, that team does not have the needed access. @Xartos or @robinelastisys can you fix that?
@elastisys/goto-public-docs do you think it makes sense to make you codeowners of this?

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:21]: No it seems to just reference the old PSPs. So it can be removed. But I think it makes more sense to remove it in a separate PR. Will fix that later.

**Comment by viktor-f** [comment]
I have made a lot of updates now and I think I have addressed all comments. PTAL

File: .github/CODEOWNERS
Comment (Xartos) [inline .github/CODEOWNERS:153]: Should this be:
```suggestion
/helmfile.d/values/minio.yaml.gotmpl @elastisys/goto-appdev-experience 
```
Since it's only used for local dev purpose

File: .github/CODEOWNERS
Comment (Xartos) [inline .github/CODEOWNERS:173]: Nit
```suggestion

#
# /helmfile.d/
#

```

File: .github/CODEOWNERS
Comment (Xartos) [inline .github/CODEOWNERS:170]: Shouldn't these have owners? Like `goto-kubernetes` or `goto-security` maybe

File: .github/CODEOWNERS
Comment (Xartos) [inline .github/CODEOWNERS:87]: `goto-appdev-experience`?

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:153]: I think that app-dev is not the correct goto area, since that is focused on app devs that use this platform not the contributors that develop this platform.
So if we want to move it, then QA might be more appropriate.

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:87]: See my comment above about minio

File: .github/CODEOWNERS
Comment (Xartos) [reply .github/CODEOWNERS:153]: Yea, I think that makes more sense than block-storage at least since this is in fact object storage.

File: .github/CODEOWNERS
Comment (Xartos) [reply .github/CODEOWNERS:87]: I still think that we should have some ownership to the contributor experience. But that might be QA then

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:170]: It would be good with owners here as well, but I was not sure which since these files "belong" to a lot of components. 
I think `goto-security` makse some sense, since this includes the PSA labels. Not sure about `goto-kubernetes`, what was your reasoning for that?

File: .github/CODEOWNERS
Comment (Zash) [inline .github/CODEOWNERS:225]: >  Unknown owner on line 225: make sure the team @elastisys/goto-public-docs exists, is publicly visible, and has write access to the repository

**Review by Zash — COMMENTED** [review]
`/config/` seems to be the only top-level directory that is not mentioned. Should it be?

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:225]: There is another thread about this further up.

**Comment by viktor-f** [comment]
> `/config/` seems to be the only top-level directory that is not mentioned. Should it be?

I forgot to write anything about this. I think we were unsure if any goto should be put as owner of the config since it touches a lot of areas. The best I can think of now would be @elastisys/goto-pipeline-qa-release, but I'm not sure.

File: .github/CODEOWNERS
Comment (Xartos) [reply .github/CODEOWNERS:170]: I agree that it's used by many different areas. But I'm just thinking since this is basically only creating standard namespaces, the issues is around namespaces and the ones that have the best knowledge about them should be kubernetes gotos.

File: .github/CODEOWNERS
Comment (Zash) [reply .github/CODEOWNERS:225]: Sorry, saw it after I did this. And then Github stopped complaining as well.

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:153]: Ok, I'll change it.

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:170]: Ok, I see your point now. Let's go with kubernetes goto for now.

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:87]: Yeah I think QA is closest at least. I'll add it to them for now.

**Review by Xartos — APPROVED** [review]

**Review by OlleLarsson — APPROVED** [review]

**Review by AlbinB97 — APPROVED** [review]

**Review by anders-elastisys — APPROVED** [review]

**Review by Elias-elastisys — APPROVED** [review]

File: .github/CODEOWNERS
Comment (simonklb) [inline .github/CODEOWNERS:167]:
```suggestion
/helmfile.d/README.md @elastisys/goto-public-docs 
```

File: .github/CODEOWNERS
Comment (simonklb) [inline .github/CODEOWNERS:193]:
```suggestion
/restore/harbor/ @elastisys/goto-container-registry
/restore/rclone/ @elastisys/goto-security
```

File: .github/CODEOWNERS
Comment (simonklb) [inline .github/CODEOWNERS:225]: This really should have an owner since it's filled with a lot of stuff.
My candidates are:
@elastisys/goto-scripts since it touches on a lot of the local cluster setup which this team currently owns as part of /scripts + @aarnq has written a lot of it
@elastisys/goto-public-docs because it's docs

File: .github/CODEOWNERS
Comment (simonklb) [inline .github/CODEOWNERS:226]: @elastisys/goto-public-docs ?

File: .github/CODEOWNERS
Comment (simonklb) [inline .github/CODEOWNERS:224]: @elastisys/goto-pipeline-qa-release ?

File: .github/CODEOWNERS
Comment (simonklb) [inline .github/CODEOWNERS:230]: @elastisys/goto-pipeline-qa-release since they own .editorconfig ?

**Review by simonklb — APPROVED** [review]
It's fine as-is but I would like to see my comments addressed as well if it's not too much work

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:225]: I'll add goto-scripts for this one.

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:224]: I did add that previously, but after discussing with @Ajarmar we said it was a bit out of scope and didn't really belong to any goto.

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:230]: Yeah that sounds close enough, I'll add that.

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:226]: Unsure about this one, I don't really know if this would fall under public docs. But it could be the closest one.

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:167]: I do actually think this is a bit closer to scripts than public docs, so I will keep that.

Comment (simonklb) [reply .github/CODEOWNERS:167]: Fine by me!

File: .github/CODEOWNERS
Comment (simonklb) [reply .github/CODEOWNERS:224]: Sure, just thought it was good to have at least one owner per file but not a strong opinion.

File: .github/CODEOWNERS
Comment (viktor-f) [reply .github/CODEOWNERS:226]: Talked to some people now and we will add @cristiklein here

**Comment by viktor-f** [comment]
Did some (hopefully) final changes now.
Using [this tool](https://github.com/jjmschofield/github-codeowners) I found some files that were missed earlier.

File: .github/CODEOWNERS
Comment (simonklb) [inline .github/CODEOWNERS:82]: Should this not be @elastisys/goto-container-registry ?

Comment (viktor-f) [reply .github/CODEOWNERS]: Yep, copy paste error. Fixed now

**Pull request merged**
